lib/view_component/slotable_v2.rb

```diff
@@ -221,10 +221,18 @@ def set_slot(slot_name, *args, **kwargs, &block)
 
       # If class
       if slot_definition[:renderable]
-        slot._component_instance = slot_definition[:renderable].new(*args, **kwargs)
+        slot._component_instance = if kwargs.present?
+                                     slot_definition[:renderable].new(*args, **kwargs)
+                                   else
+                                     slot_definition[:renderable].new(*args)
+                                   end
```

Comment:
If there is no keyword arguments then `**{}` becomes `{}`:

```ruby
def foo; end

foo(**{}) # equivalent to `foo({})`
ArgumentError: wrong number of arguments (given 1, expected 0)
```

Reply:
Thanks for opening this!

I'm just now getting a chance to take a peek. I think the issue is how we're delegating args + kwargs and the (unfortunate) differences between Ruby versions.

In order to get things working 100% I think we can:

1. Call `ruby2_keywords(METHOD_NAME_AS_SYMBOL) if respond_to?(:ruby2_keywords, true)` on all methods that currently accept `**kwargs` (including methods defined via `define_method`)
2. Remove all uses of `**kwargs` and forward only `*args` since the `ruby2_keywords` method should ensure that kwargs work for 2.7+

If you'd like to give that a try, that would be awesome. No pressure though, I'm happy to make that change and push it up to this branch if you'd like!

Reply:
Ok so it seems to work like a charm! 👍 Tested with ruby 2.6.7 and 2.7.

I've reverted rails downgrade. Do you want me to clean the branche history?

Reply:
It's failing with Ruby 3.0 and I don't find how to fix it. Any clue?

Reply:
Sorry for the delayed response, I've been very busy this week.

> I've reverted rails downgrade. Do you want me to clean the branche history?
No need, we're not strict about commit history in this project, but thank you for offering!

>It's failing with Ruby 3.0 and I don't find how to fix it. Any clue?

Not totally sure yet, I'll try to find some time this week to take a peek and see if anything stands out.